### PR TITLE
Add libgpod for mac

### PR DESCRIPTION
**Pull request by jbroadus**
This change was taken from Strawberry. When and if a homebrew libgpod package becomes available, this change can be replaced.

Reference: strawberrymusicplayer/strawberry@f2c7df3

### CODE REVIEW

**Comment by jbroadus** [comment]
Follow-up change for #7004

File: .github/workflows/all.yml [hunk 1]
```diff
@@ -890,6 +890,31 @@ jobs:
...
+          patch -p0 < patch-tools-generic-callout.c.diff
+          patch -p1 < libgpod-libplist.patch
+          chmod u+x autogen.sh
+          ./autogen.sh
+          ./configure --disable-more-warnings \
+                      --disable-silent-rules \
+                      --disable-udev \
+                      --disable-pygobject \
+                      --with-python=no
+          make -j8
```
Comment (hatstand) [inline .github/workflows/all.yml:915]: `-j8` is optimistic on the instances we use here. `-j2` makes more sense.

File: Brewfile [hunk 2]
```diff
@@ -17,3 +17,14 @@ brew 'protobuf'
 brew 'protobuf-c'
 brew 'qt@5'
 brew 'tomahawk-player/homebrew-tomahawkqt5/liblastfm'
+
+# These were added for libgpod support and can probably be removed if/when the
+# libgpod package is available.
+brew 'wget'
```
Comment (hatstand) [inline Brewfile:23]: alpha-sort

File: .github/workflows/all.yml (hunk 1 above)
Comment (jbroadus) [reply .github/workflows/all.yml:915]: Yeah, this was cut-and-paste from Strawberry. I'll fix.

File: Brewfile (hunk 2 above)
Comment (jbroadus) [reply Brewfile:27]: Done

File: Brewfile [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
 brew 'glib'
+brew 'gnome-common'
 brew 'google-sparsehash'
 brew 'gst-libav'
 brew 'gst-plugins-bad'
 brew 'gst-plugins-base'
 brew 'gst-plugins-good'
 brew 'gst-plugins-ugly'
 brew 'gstreamer'
+brew 'gtk-doc'
```
Comment (jbroadus) [inline Brewfile:18]: I'm not sure why this is needed. Running a test to see if it can be removed.

**Pull request merged**